File: CallKit/CallKitObjCSample/CallKit/OutgoingCallViewController.m
//
//  OutgoingCallViewController.m
//  CallKit
//
//  Created by Dobrinka Tabakova on 9/26/16.
//  Copyright © 2016 Dobrinka Tabakova. All rights reserved.
//

#import "OutgoingCallViewController.h"
#import "CallViewController.h"


static NSString *const kTitle = @"Dial Call";

@interface OutgoingCallViewController ()
@property (weak, nonatomic) IBOutlet UIButton *dialButton;
@property (weak, nonatomic) IBOutlet UITextField *phoneNumberTextField;

@end

@implementation OutgoingCallViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = kTitle;
}

#pragma mark - Actions

- (IBAction)phoneNumberValueChanged:(UITextField*)sender {
    NSString *phoneNumber = [sender.text stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
    self.dialButton.enabled = (phoneNumber.length);
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier] isEqualToString:@"ShowDial"]) {
        CallViewController* callViewController = [segue destinationViewController];
        callViewController.phoneNumber = self.phoneNumberTextField.text;
    }
}

@end

File: CallKit/CallKitSwiftSample/CallKitSwiftSample/OutgoingCallViewController.swift
//
//  OutgoingCallViewController.swift
//  CallKitSwiftSample
//
//  Created by Nikolay Andonov on 11/15/16.
//  Copyright © 2016 Mentormate. All rights reserved.
//

import UIKit

class OutgoingCallViewController: UIViewController {
    
    private let kTitle = "Dial Call"
    
    @IBOutlet weak var phoneNumberTextField: UITextField!
    @IBOutlet weak var dialButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        title = kTitle
    }

    // MARK: - Actions

    @IBAction func phoneNumberValueChanged(_ sender: UITextField) {
        
        let phoneNumber = sender.text?.trimmingCharacters(in: .whitespacesAndNewlines)
        dialButton.isEnabled = (phoneNumber?.characters.count)! > 0
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        
        if segue.identifier == "ShowDial" {
            let callViewController = segue.destination as! CallViewController
            callViewController.phoneNumber = phoneNumberTextField.text
        }
    }
}
